stock_packaging_calculator/models/product.py

```diff
@@ -118,7 +118,12 @@ def _product_qty_by_packaging(self, pkg_by_qty, qty, with_contained=False):
             "_packaging_values_handler", self._prepare_qty_by_packaging_values
         )
         for pkg in pkg_by_qty:
-            qty_per_pkg, qty = self._qty_by_pkg(pkg.qty, qty)
+            # Boost perf: no need to deduce the qty_per_pkg if the pkg_qty is 1
+            if float_compare(pkg.qty, 1, precision_digits=self.uom_id.rounding) == 0:
```

Comment:
```suggestion
            if float_compare(qty, 1, precision_digits=self.uom_id.rounding) == 0:
```
I don't know for sure just base on reading seems another qty needs to be used

Reply:
The code was right, `qty` is the whole qty (e.g. 150 000) while `pkg.qty` is the packaging qty (e.g. 1), and the `_qty_by_pkg()` method subtract `pkg.qty` from `qty`:
```python
    def _qty_by_pkg(self, pkg_qty, qty):
        """Calculate qty needed for given package qty."""
        qty_per_pkg = 0
        while (
            float_compare(qty - pkg_qty, 0.0, precision_digits=self.uom_id.rounding)
            >= 0.0
        ):
            qty -= pkg_qty
            qty_per_pkg += 1
        return qty_per_pkg, qty
```

Reply:
can we have multiple pkg_qties in pkg_by_qty?
If not, why not using divmod?

Reply:
I didn't understood your questions?

Reply:
I don't know what represents pkg_by_qty, if qty is 1000, and pkg_qty is 20, then the result is qty % pkg_qty.
Might save some time for other cases too, no?

Reply:
Maybe, I didn't analyzed the code too much so that's a question for @simahawk.
I didn't wanted to refactor the whole thing and wanted to avoid the simplest case where the pkg.qty is 1, making the while loop useless.